src/lib/precomputed_values/precomputed_values.ml

```diff
@@ -0,0 +1,81 @@
+open Core
+module T = Genesis_proof.T
+include T
+
+let hashes =
+  lazy
+    (let constraint_constants =
+       Genesis_constants.Constraint_constants.compiled
+     in
+     let proof_level = Genesis_constants.Proof_level.compiled in
+     let ts =
+       Transaction_snark.constraint_system_digests ~constraint_constants ()
+     in
+     let bs =
+       Blockchain_snark.Blockchain_snark_state.constraint_system_digests
+         ~proof_level ~constraint_constants ()
+     in
+     ts @ bs )
+
+let for_unit_tests =
+  lazy
+    (let open Staged_ledger_diff in
+    let protocol_state_with_hashes =
+      Mina_state.Genesis_protocol_state.t
+        ~genesis_ledger:
+          (let open Genesis_ledger in
+          Packed.t for_unit_tests)
+        ~genesis_epoch_data:Consensus.Genesis_epoch_data.for_unit_tests
+        ~constraint_constants:
+          Genesis_constants.Constraint_constants.for_unit_tests
+        ~consensus_constants:(Lazy.force Consensus.Constants.for_unit_tests)
+        ~genesis_body_reference
+    in
+    { runtime_config = Runtime_config.default
+    ; constraint_constants =
+        Genesis_constants.Constraint_constants.for_unit_tests
+    ; proof_level = Genesis_constants.Proof_level.for_unit_tests
+    ; genesis_constants = Genesis_constants.for_unit_tests
+    ; genesis_ledger = Genesis_ledger.for_unit_tests
+    ; genesis_epoch_data = Consensus.Genesis_epoch_data.for_unit_tests
+    ; genesis_body_reference
+    ; consensus_constants = Lazy.force Consensus.Constants.for_unit_tests
+    ; protocol_state_with_hashes
+    ; constraint_system_digests =
+        lazy
+          (List.map (Lazy.force hashes) ~f:(fun (x, y) ->
```

Comment:
This `List.map` is now not useful, you can just replace the whole expressions with `hashes`.

Reply:
should have cranked that reduction machine a little harder :grimacing: